### PR TITLE
Port meta-mender to Yocto daisy branch

### PR DESCRIPTION
**Pull request by kacf**

### CODE REVIEW

**Comment by kacf** [comment]
@pasinskim: Can you take a look?

File: classes/mender-image-buildinfo.bbclass [hunk 1]
```diff
@@ -1,15 +1,13 @@
-inherit image-buildinfo
-
 DEVICE_TYPE ?= "${MACHINE}"
 IMAGE_ID = "${PN}-${DATETIME}"
-IMAGE_BUILDINFO_VARS = "DISTRO DATETIME PN IMAGE_ID DEVICE_TYPE"
 
 buildinfo_mender () {
 cat > ${IMAGE_ROOTFS}${sysconfdir}/build_mender << END
```
Comment (pasinskim) [inline classes/mender-image-buildinfo.bbclass]: If we are changing this maybe we  can change the horrible name of this file. But I guess we can do it in separate PR.

File: classes/sdimg.bbclass [hunk 2]
```diff
@@ -39,21 +42,24 @@ IMAGE_BOOT_ENV_FILE ?= "uboot.env"
 # u-boot environment file
 IMAGE_UENV_TXT_FILE ?= "uEnv.txt"
 
+# These are usually defined in the machine section, but for daisy this
+# concept doesn't exist yet.
+IMAGE_BOOT_FILES ?= "u-boot.${UBOOT_SUFFIX}"
+IMAGE_BOOT_FILES_append_beaglebone ?= " MLO"
```
Comment (pasinskim) [inline classes/sdimg.bbclass]: Shouldn't we have `IMAGE_BOOT_FILES_append_beaglebone = " MLO"`?

File: classes/sdimg.bbclass [hunk 3]
```diff
@@ -100,20 +103,88 @@ IMAGE_CMD_sdimg() {
...
+        # COMMIT changes to image file
+        echo p
+        echo w
+
+    ) | fdisk -c=nondos -u=sectors $SDIMG
+
+    dd if=${WORKDIR}/fat.dat of=$SDIMG seek=$PART1_START_SECTORS conv=notrunc
+    dd if=${WORKDIR}/part1.tmp of=$SDIMG seek=$PART2_START_SECTORS conv=notrunc
+    dd if=${WORKDIR}/part2.tmp of=$SDIMG seek=$PART3_START_SECTORS conv=notrunc
+    dd if=${WORKDIR}/data.ext3 of=$SDIMG seek=$PART4_START_SECTORS conv=notrunc
```
Comment (pasinskim) [inline classes/sdimg.bbclass]: You did a nice cleanup with the names on master branch. Maybe something similar here? `BOOT_PART_START`, `INACTIVE_START`, and so on?

Comment (kacf) [inline classes/sdimg.bbclass]: You're right, I should do that here as well.

File: classes/sdimg.bbclass (hunk 2 above)
Comment (kacf) [inline classes/sdimg.bbclass]: Hmm, my impression is that it's better to leave things so they can be overridden by users of the layer. I'm unsure what's the best practice here. @bboozzoo: Do you have any advice?

File: classes/mender-image-buildinfo.bbclass (hunk 1 above)
Comment (kacf) [inline classes/mender-image-buildinfo.bbclass]: Yes, that's preexisting, I agree it has a horrible name, but I won't handle that in the PR.

File: classes/sdimg.bbclass (hunk 2 above)
Comment (bboozzoo) [inline classes/sdimg.bbclass]: IIRC `SOMEVAR ?= "foo"` makes sense for assignments only.

Make sure to keep the append as `SOMEVAR_append_beaglebone` (`beaglebone` coming in after `append`) though.

Comment (kacf) [inline classes/sdimg.bbclass]: Ok, I will switch to `=` then.

**Comment by kacf** [comment]
Issues fixed!
